pkg/interfaces/contracts/pool-weighted/ILBPMigrationRouter.sol

```diff
@@ -0,0 +1,104 @@
+
+/// @notice Interface for migrating liquidity from an LBP to a new Weighted Pool with custom parameters.
+interface ILBPMigrationRouter {
+    /**
+     * @notice The pool was successfully migrated from an LBP to a new weighted pool.
+     * @param lbp The LB Pool that was migrated
+     * @param weightedPool The newly created weighted pool
+     * @param bptAmountOut The amount of BPT tokens received from the weighted pool after migration
+     */
+    event PoolMigrated(ILBPool indexed lbp, IWeightedPool weightedPool, uint256 bptAmountOut);
+
+    /**
+     * @notice A contract returned from the Balancer Contract Registry is not active.
+     * @param contractName The name of the contract that is not active
+     */
+    error ContractIsNotActiveInRegistry(string contractName);
+
+    /**
+     * @notice `migrateLiquidity` was called before the sale completed.
+     * @param lbp The Liquidity Bootstrapping Pool with unfinalized weights
+     */
+    error LBPWeightsNotFinalized(ILBPool lbp);
+
+    /**
+     * @notice Input amount of a token is less than existing amounts in the LBP.
+     * @param token The token with an insufficient input amount
+     * @param actualAmount The actual amount of the token provided
+     */
+    error InsufficientInputAmount(IERC20 token, uint256 actualAmount);
+
+    /**
+     * @notice The caller is not the owner of the LBP.
+     */
+    error SenderIsNotLBPOwner();
+
+    struct MigrationHookParams {
+        ILBPool lbp;
+        IWeightedPool weightedPool;
+        IERC20[] tokens;
+        address sender;
+        address excessReceiver;
+        uint256[] exactAmountsIn;
+        uint256 minAddBptAmountOut;
+        uint256[] minRemoveAmountsOut;
+    }
+
+    struct WeightedPoolParams {
+        string name;
+        string symbol;
+        uint256[] normalizedWeights;
+        PoolRoleAccounts roleAccounts;
+        uint256 swapFeePercentage;
+        address poolHooksContract;
+        bool enableDonation;
+        bool disableUnbalancedLiquidity;
+        bytes32 salt;
+    }
+
+    /**
+     * @notice Migrates liquidity from an LBP to a new weighted pool with custom parameters.
+     * @param lbp Liquidity Bootstrapping Pool
+     * @param exactAmountsIn The exact amounts of each token to add to the weighted pool
+     * @param minAddBptAmountOut Minimum amount of BPT tokens expected to receive
+     * @param minRemoveAmountsOut Minimum token amounts expected when removing from the LBP
+     * @param excessReceiver Address to receive excess tokens after migration
+     * @param params Parameters for creating the new weighted pool
+     * @return weightedPool The newly created weighted pool
+     * @return bptAmountOut The amount of BPT tokens received from the weighted pool after migration
+     */
+    function migrateLiquidity(
+        ILBPool lbp,
+        uint256[] memory exactAmountsIn,
+        uint256 minAddBptAmountOut,
+        uint256[] memory minRemoveAmountsOut,
```

Comment:
As discussed off-thread, the migration parameter will be a maximum liquidity percentage.

About the slippage, it shouldn't really be necessary, and this is going to be called by another contract in the end. So maybe a single FP-18 number representing the maximum BPT percentage to migrate should do.

The router will exit all of the liquidity, and initialize the new pool with a percentage of it keeping the same price. The rest goes to the `excessReceiver` address.